sprockets/mixins/amqp/amqp.py

```diff
@@ -0,0 +1,199 @@
+
+    def _create_new_channel(self):
+        """Open a channel on the class connection"""
+        AMQPMixin.connection.channel(on_open_callback=self.on_channel_open)
+
+    def _new_message_properties(self, content_type=None, content_encoding=None,
+                                headers=None, delivery_mode=None,
+                                priority=None, correlation_id=None,
+                                reply_to=None, expiration=None,
+                                message_id=None, timestamp=None,
+                                message_type=None, user_id=None, app_id=None):
+        """Create a BasicProperties object, with the properties specified
+
+        :param str content_type: MIME content type
+        :param str content_encoding: MIME content encoding
+        :param dict headers: Message header field table
+        :param int delivery_mode: Non-persistent (1) or persistent (2)
+        :param int priority: Message priority, 0 to 9
+        :param str correlation_id: Application correlation identifier
+        :param str reply_to: Address to reply to
+        :param str expiration: Message expiration specification
+        :param str message_id: Application message identifier
+        :param int timestamp: Message timestamp
+        :param str message_type: Message type name
+        :param str user_id: Creating user id
+        :param str app_id: Creating application id
+        :rtype: pika.BasicProperties
+
+        """
+        return pika.BasicProperties(content_type, content_encoding, headers,
+                                    delivery_mode, priority, correlation_id,
+                                    reply_to, expiration, message_id,
+                                    timestamp, message_type, user_id, app_id)
+
+    def _publish_message(self, exchange, routing_key, message, properties):
+        """Publish the message to RabbitMQ
+
+        :param str exchange: The exchange to publish to
+        :param str routing_key: The routing key to publish with
+        :param str message: The message body
+        :param pika.BasicProperties: The message properties
+
+        """
+        LOGGER.debug('Publishing message: %s to exchange: %s with '
+                     'routing_key: %s and properties: %s',
+                     message, exchange, routing_key, properties)
+        if not self.connection or not self.channel:
+            LOGGER.error('Unable to connect to Rabbit')
+            raise pika.exceptions.AMQPError()
+        AMQPMixin.channel.basic_publish(exchange, routing_key, message,
+                                        properties)
+
+    @property
+    def _rabbit_params(self):
+        """Return a pika URLParameters object using the :envvar:`AMQP`
+        environment variable which identifies the Rabbit MQ server as a URL
+        ready for :class:`pika.connection.URLParameters`.
+
+        :rtype: pika.URLParameters
+
+        """
+        LOGGER.debug('URLParameters: %s',
```

Comment:
Won't this blow up with a ``KeyError`` if the environment variable is not set? And why create two objects, just for logging?  How about something like:

```python
params = pika.URLParameters(os.environ.get('AMQP', 'amqp://guest:guest@localhost:5672/%2f'))
LOGGER.debug('RabbitMQ connection params: %r', params)
return params
```